src/irr/video/COpenGLDescriptorSet.h

```diff
@@ -125,21 +130,26 @@ class COpenGLDescriptorSet : public IGPUDescriptorSet
             }
         };
         size_t u=0ull, s=0ull, t=0ull, i=0ull;
+        size_t dyn_offset_iter = 0ull;
         for (const auto& desc : (*m_descriptors))
         {
-            if (desc.descriptorType==asset::EDT_UNIFORM_BUFFER)
+            if (desc.descriptorType==asset::EDT_UNIFORM_BUFFER || desc.descriptorType==asset::EDT_UNIFORM_BUFFER_DYNAMIC)
                 for (const auto& info : (*desc.info)) {
+                    const uint32_t extra_offset = (desc.descriptorType==asset::EDT_UNIFORM_BUFFER_DYNAMIC) ? (*m_dynamicOffsets)[dyn_offset_iter] : 0u;
```

Comment:
you apply the extra offset when binding the descriptor set in `COpenGLDriver` not here, because the dynamic offset is not part of the descriptor set state!

Reply:
@Crisspl !?

Reply:
resolved